src/java/htsjdk/samtools/SAMTextWriter.java

```diff
@@ -80,8 +76,42 @@ public Writer getWriter() {
      * @param stream Need not be buffered because this class provides buffering. 
      */
     public SAMTextWriter(final OutputStream stream) {
+        this(stream, SamFlagField.DEFAULT);
+    }
+
+    /**
+     * Constructs a SAMTextWriter that outputs to a Writer.
+     * @param out Writer.
+     */
+    public SAMTextWriter(final Writer out, final SamFlagField samFlagFieldOutput) {
+        this.out = out;
+        this.file = null;
+        this.samFlagFieldOutput = samFlagFieldOutput;
+    }
+
+    /**
+     * Constructs a SAMTextWriter that writes to a File.
+     * @param file Where to write the output.
+     */
+    public SAMTextWriter(final File file, final SamFlagField samFlagFieldOutput) {
+        try {
+            this.file = file;
+            this.out = new AsciiWriter(new FileOutputStream(file));
+        } catch (final IOException e) {
+            throw new RuntimeIOException(e);
+        }
+        this.samFlagFieldOutput = samFlagFieldOutput;
+    }
+
+    /**
+     * Constructs a SAMTextWriter that writes to an OutputStream.  The OutputStream
+     * is wrapped in an AsciiWriter, which can be retrieved with getWriter().
+     * @param stream Need not be buffered because this class provides buffering.
+     */
+    public SAMTextWriter(final OutputStream stream, final SamFlagField samFlagFieldOutput) {
         this.file = null;
         this.out = new AsciiWriter(stream);
+        this.samFlagFieldOutput = samFlagFieldOutput;
```

Comment:
I will do it, but the other objects are not being checked for null here, so it seems like overkill.